google/google-how-to/gce/set-hostname-using-cloudinit.rst

```diff
@@ -1,20 +1,29 @@
 Set hostname of GCE instances
-=================================
+=============================
 
-The hostname of GCE instances can be set using multiple methods. Apart from setting it manually, one option is to use a set-up tool like cloud-init, while Google's preferred method is to use its DHCP service. Both these options are considered here.
+The hostname of GCE instances can be set using multiple methods. Google's preferred method is to use its DHCP service, which requires you to choose a fully qualified DNS name (FQDN), e.g. something like test123.test.com. If you don't want an FQDN or if you want a specific format for consistent naming across clouds, you can use a set-up tool like cloud-init to set your hostname.
 
-.. _use_cloud-init:
+Both these methods are described here. Also, due to a recent hostname-related update on GCP, you might have to make some additional changes for GCE images that use Ubuntu 24.04 and later. These are explained at the end.
 
-Option 1: Use cloud-init
-------------------------
 
-You can set the required hostname by programming the cloud-init metadata. For this, use the `gcloud compute instances add-metadata`_ command:
+Using DHCP (Google's preferred method)
+--------------------------------------
+
+By default, Google's DHCP service sets the hostname to an automatically generated internal DNS name. 
+
+To set your own custom name, follow the instructions given in `Create a VM instance with a custom hostname`_. In this case, the DHCP service will additionally provide the custom name and will prioritise it to be the default hostname. However, as mentioned earlier, the custom name needs to be an FQDN and you also need to create a corresponding DNS record for it.
+
+
+Using cloud-init
+----------------
+
+cloud-init uses the ``hostname`` command to programmatically set the hostname. You just need to configure its metadata with the required hostname and use the `gcloud compute instances add-metadata`_ command:
 
 .. code::
 
     gcloud compute instances add-metadata INSTANCE_NAME --metadata-from-file=KEY=VALUE
 
-Here INSTANCE_NAME is your VM name and the metadata is specified using a KEY=VALUE pair, e.g. user-data=FILENAME, where FILENAME is the local path to a file that contains the desired cloud-init configurations (including the hostname). Specify the required hostname in the user-data file:
+Here INSTANCE_NAME is your VM name and the metadata is specified using a KEY=VALUE pair. For instance, the metadata could be specified as 'user-data=FILENAME', where FILENAME is the local path to a file that contains the desired cloud-init configurations. Include the desired hostname in that user-data file:
 
 .. code:: 
 
```

Comment:
Probably we should have this set a proper userdata file, e.g.
```
#cloud-config

hostname: test123
```

Reply:
Done